api/src/commonMain/kotlin/dev/kord/cache/api/observables/DataCache.kt

```diff
@@ -0,0 +1,59 @@
+package dev.kord.cache.api.observables
+
+/**
+ * A cache for a map of entries where each [Value] is associated with a unique [Key].
+ * */
+interface DataCache<Key : Any, Value : Any> {
```

Comment:
a function that's still missing is something that can return multiple values, i suggest a signature like:
```kt
fun filter(predicate: (Value) -> Boolean): Flow<Value>
```